examples/lambda/lambda/main.tf

```diff
@@ -0,0 +1,57 @@
+locals {
+  tags = merge(
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/enabled" : "true" },
+    var.consul_datacenter != "" ?
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/datacenter" : var.consul_datacenter } : {},
+    var.consul_partition != "" ?
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/partition" : var.consul_partition } : {},
+    var.consul_namespace != "" ?
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/namespace" : var.consul_namespace } : {},
+    var.payload_passthrough != "" ?
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/payload-passthrough" : var.payload_passthrough } : {},
+    length(var.aliases) > 0 ?
+    { "serverless.consul.hashicorp.com/v1alpha1/lambda/aliases" : join("+", var.aliases) } : {}
+  )
+
+  layers = concat(
+    var.layers,
+    var.consul_lambda_extension_arn != "" ? [var.consul_lambda_extension_arn] : [],
+  )
+}
+
+resource "aws_lambda_function" "lambda_app" {
```

Comment:
I'm thinking that maybe this module can form the basis of a `mesh-function` module that we provide for end users, analogous to `mesh-task` for ECS.